cmake/BLTInstallableMacros.cmake

```diff
@@ -428,37 +428,143 @@ endmacro(blt_add_target_compile_flags)
+        list(APPEND _depends_on "${_BLT_${_target_upper}_DEPENDS_ON}")
+    endif()
+
+    # Get dependencies if CMake target
+    if(TARGET ${arg_TARGET})
+        get_property(_target_type TARGET ${arg_TARGET} PROPERTY TYPE)
+        if(NOT "${_target_type}" STREQUAL "INTERFACE_LIBRARY")
+            get_target_property(_propval ${arg_TARGET} LINK_LIBRARIES)
+            if(_propval)
+                list(APPEND _depends_on ${_propval})
+            endif()
         endif()
+
+        # get interface link libraries
+        get_target_property(_propval ${arg_TARGET} INTERFACE_LINK_LIBRARIES)
+        if (_propval)
+            list(APPEND _depends_on ${_propval})
+        endif()
+    endif()
+
+    blt_list_remove_duplicates(TO _depends_on)
+    foreach(t ${_depends_on})
+        if (NOT "${t}" IN_LIST ${arg_TLIST})
+            list(APPEND ${arg_TLIST} ${t})
+            blt_find_target_dependencies(TARGET ${t} TLIST ${arg_TLIST})
+        endif()
+    endforeach()
+
+    unset(_depends_on)
+endmacro(blt_find_target_dependencies)
+
+
+##------------------------------------------------------------------------------
+## blt_convert_to_system_includes(TARGETS [<target>...]
+##                                CHILDREN [TRUE|FALSE]
+##                                [QUIET])
+##
+## Converts existing interface includes to system interface includes.
+##------------------------------------------------------------------------------
+function(blt_convert_to_system_includes)
+    set(options QUIET)
+    set(singleValuedArgs CHILDREN TARGET)
+    set(multiValuedArgs TARGETS)
+
+    ## parse the arguments to the macro
+    cmake_parse_arguments(arg
+        "${options}" "${singleValuedArgs}" "${multiValuedArgs}" ${ARGN})
+
+    if(DEFINED arg_TARGET)
+        message(WARNING "TARGET is a deprecated parameter for the blt_convert_to_system_includes macro.")
+        list(APPEND arg_TARGETS ${arg_TARGET})
     endif()
 
-    unset(_include_dirs)
-endmacro()
+    if(NOT DEFINED arg_TARGETS)
+        message(FATAL_ERROR "TARGETS is a required parameter for the blt_convert_to_system_includes macro.")
+    endif()
+
+    if(NOT DEFINED arg_CHILDREN)
+        set(arg_CHILDREN TRUE)
+    endif()
+
+    # PGI does not support -isystem
+    if(NOT "${CMAKE_CXX_COMPILER_ID}" STREQUAL "PGI")
+        set(_target_list "")
+
+        foreach(_target ${arg_TARGETS})
+            if(TARGET ${_target})
+                if(${arg_CHILDREN})
+                    blt_find_target_dependencies(TARGET ${_target} TLIST _target_list)
+                endif()
+
+                list(APPEND _target_list ${_target})
+            elseif(NOT ${arg_QUIET)
+                message(WARNING "${_target} does not exist!")
+            endif()
+        endforeach()
+
+        blt_list_remove_duplicates(TO _target_list)
+
+        foreach(_target ${_target_list})
+            # Ignore entries that are not actual CMake targets
+            if(TARGET ${_target})
```

Comment:
Do you know how I would update include directories on a BLT registered target?

Reply:
Would it be as simple as `set(_BLT_${uppercase_dependency}_TREAT_INCLUDES_AS_SYSTEM TRUE)`?

Reply:
that is correct